MovingBlocks/Terasology
# Fix normal maps when looking at a block on all sides.

### Contains

- Fix normal maps when looking at a block on all sides.
- Add "extra lighting" video option to turn on normal maps.
- Add gloss map block tiles that turn on specular light reflections for parts of a tile. This replaces the ```LightComponent.lightSpecularIntensity``` field.
- Add IronBar block to demonstrate the gloss effect.

[Video demo](https://youtu.be/54b7kA25Vno)

There is one ill side effect of this change in that the normal world atlas image that is spit out into the screenshots directory will look mostly white because by default it is fully transparent.

### How to test

Turn on "extra lighting" in the video options.  Ensure that specular lighting (shininess/glossiness) does not appear by default on blocks.  Place a IronBar block and hold up a torch to it.

### Outstanding before merging
- [x] Adjust the lightProperties to be a vec3 instead of a vec4 with an unused field.
- [x] Rename applyGrayscaleMaskToAlpha(...) to storeGreyscaleMapIntoAlpha(...) and make the parameter names more clear.
